moving_modules.md

```diff
@@ -0,0 +1,224 @@
+
+The following types are internal to Puppet and will remain as is:
+
+    component
+    schedule
+    stage
+    whit
+
+## Core
+
+The following types will be left in Puppet for now. The `file`, `user`, and `group` types are needed to apply settings catalogs. The `filebucket`, `resources`, and `tidy` types know too much about puppet internals to be removed. The `notify` type is used extensively in puppet rspec tests as it is the most basic (providerless) type. The `package` and `service` types have multiple providers for each type, which makes removal more difficult. We may remove them at a later time (TBD).
+
+    exec
+    file
+    filebucket
+    group
+    notify
+    package
+    resources
+    service
+    tidy
+    user
+
+## External
+
+The following types and providers will be removed from Puppet. A subset (details TBD below) will be added back to puppet-agent during packaging.
+
+Each top-level path below specifies the name of the module, e.g. `augeas`, and the files contained within each module. The modules will be installed in a new directory visible to Puppet's autoloader, so catalog compilation and application will just work without additional configuration. Puppet will prefer modules in the modulepath and pluginsync'ed lib directory over the packaged modules, so that newer versions of modules can fix bugs in packaged modules.
+
+    Path                                           Comments
+
+    <vendored_modules>/                            (*nix) /opt/puppetlabs/puppet/vendored_modules/
+                                                   (Windows) C:\Program Files\Puppet Labs\Puppet\puppet\vendored_modules
+
+      augeas/                                      Depends on 'puppet/parameter/boolean'
+        lib/puppet/feature/augeas.rb               Extracted from lib/puppet/features/base.rb        
+        lib/puppet/type/augeas.rb
+        lib/puppet/provider/augeas/augeas.rb
+
+      cron/                                        Depends on 'puppet/provider/parsedfile',
+        lib/puppet/type/cron.rb                               'puppet/util/filetype'
+
+      host/                                        Depends on 'puppet/property/ordered_list',
+        lib/puppet/type/host.rb                               'puppet/provider/parsedfile',
+        lib/puppet/provider/host/parsed.rb
+
+      k5login/                                     Depends on 'puppet/type/file/selcontext',
+        lib/puppet/type/k5login.rb                            'puppet/util/selinux'
+                                                              
+      mailalias/                                   Depends on 'puppet/provider/parsedfile'
+        lib/puppet/type/mailalias.rb
+        lib/puppet/provider/mailalias/aliases.rb
+     
+      maillist/
+        lib/puppet/type/maillist.rb
+        lib/puppet/provider/maillist/maillist.rb
+
+      macdslocal/                                  Depends on 'puppet/provider/nameservice/directory_service',
```

Comment:
i think `provider/nameservice/directory_service` should come along for the ride here

Reply:
Unfortunately that provider is the parent for user and group on osx also, and moving those out is complicated by need to manage settings catalogs and that the resources have multiple providers...

Reply:
Is the name of the module ok? Wasn't sure if it should be `dslocal` or `osxdslocal`, etc..?